flink-runtime/src/main/java/org/apache/flink/runtime/state/delegate/DelegateStateBackend.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.delegate;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.runtime.state.StateBackend;
+
+/**
+ * An interface for delegate state backend.
+ *
+ * <p>As its name, it should include a state backend to delegate. The delegated state backend itself
+ * can not be a {@link DelegateStateBackend}.
+ *
+ * <p>TODO: provide configurable way to verify whether a state backend can be delegated.
+ */
+@Internal
+public interface DelegateStateBackend extends StateBackend {
```

Comment:
`DelegateStateBackend` -> `DelegatingStateBackend`?
Just "Delegate" is ambiguous to me.

Reply:
For an "object", I'd prefer not to use "ing" format; it causes more confusion.

Reply:
I don't see any confusion with "ing".

But putting this aside, 
using this suffix is quite common in general, in Flink and even in this PR: `ChangelogReducingState`, `ChangelogAggregatingState`.

So I'd name it `DelegatingStateBackend` at least for consistency.

Reply:
They are different. 
First,
XXX Reducing State, reducing is quite obviously an adj-noun;

while

DelegatingStateBackend, I can not tell it is an adj-noun or action;

Second,
I feel "Reducing" and "Aggregating" themselves are confusing as well; For Map state, should we call it Mapping State?